app/models/concerns/wakes/metrics/facebook.rb

```diff
@@ -7,13 +7,26 @@ module Facebook
       included do
         store_accessor :document, :facebook_count, :facebook_count_updated_at
 
+        before_validation :prevent_update_of_legacy_location, on: :update
+
         def enqueue_facebook_count_update
           Wakes::UpdateFacebookMetricsJob.perform_later(self)
         end
 
         def self.ordered_for_facebook_updates
           order("document->'facebook_count_updated_at' ASC NULLS FIRST")
         end
+
+        protected
+
+        def prevent_update_of_legacy_location
+          throw :abort if facebook_count_will_change? && !canonical
```

Comment:
@benhutton Yes, this is [the Rails 5 way](http://blog.bigbinary.com/2016/02/13/rails-5-does-not-halt-callback-chain-when-false-is-returned.html) of halting the callback chain when a validation fails (returning `false` no longer works).